libs/s3/s3_test_lib.py

```diff
@@ -640,19 +640,24 @@ def get_object(
             bucket: str = None,
             key: str = None,
             ranges: str = None,
-            raise_exec: bool = True) -> tuple:
+            raise_exec: bool = True,
+            skip_polling: bool = False) -> tuple:
         """
         Retrieve object from specified S3 bucket.
 
         :param raise_exec: raise an exception in default case.
         :param key: Key of the object to get.
         :param ranges: Byte range to be retrieved
         :param bucket: The bucket name containing the object.
+        :param skip_polling: Skip retry for GET Object, in case of failures
         :return: (Boolean, Response)
         """
         try:
             LOGGER.info("Retrieving object from a bucket")
-            response = poll(super().get_object, bucket, key, ranges)
+            if not skip_polling:
```

Comment:
Is this expected in RGW?

Reply:
Polling was causing issues in test code which ends up waiting for failure while the lib function is polling for success unnecessarily, so added an option to bypass it